src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/RC2.cs

```diff
@@ -0,0 +1,77 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Internal.Cryptography;
+using System.ComponentModel;
+
+namespace System.Security.Cryptography
+{
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public abstract class RC2 : SymmetricAlgorithm
+    {
+        protected int EffectiveKeySizeValue;
+
+        protected RC2()
+        {
+            LegalBlockSizesValue = s_legalBlockSizes;
+            LegalKeySizesValue = s_legalKeySizes;
+            KeySize = 128;
+            BlockSize = 64;
+        }
+
+        public static RC2 Create()
+        {
+            return new RC2Implementation();
+        }
+
+        public override int KeySize
+        {
+            get { return KeySizeValue; }
+            set
+            {
+                if (value < EffectiveKeySizeValue)
+                    throw new CryptographicException(SR.Cryptography_RC2_EKSKS);
+
+                base.KeySize = value;
+            }
+        }
+
+        public virtual int EffectiveKeySize
+        {
+            get
+            {
+                if (EffectiveKeySizeValue == 0)
+                    return KeySizeValue;
+
+                return EffectiveKeySizeValue;
+            }
+            set
+            {
+                if (value > KeySizeValue)
+                    throw new CryptographicException(SR.Cryptography_RC2_EKSKS);
+                else if (value == 0)
+                    EffectiveKeySizeValue = value;
+                else if (value < 40)
+                    throw new CryptographicException(SR.Cryptography_RC2_EKS40);
+                else
+                {
+                    if (value.IsLegalSize(s_legalKeySizes))
+                        EffectiveKeySizeValue = value;
+                    else
+                        throw new CryptographicException(SR.Cryptography_InvalidKeySize);
+                }
+            }
+        }
```

Comment:
There are a lot of branches in this setter, and I don't know that they're all tested since the implementations require that KeySizeValue be EffectiveKeySizeValue (so the backwards-looking first if never gets hit... I think).

I'm also not sure that it can actually be reset to 0.